sdv/data_processing/data_processor.py

```diff
@@ -571,7 +572,13 @@ def _create_config(self, data, columns_created_by_constraints):
                 else:
                     bothify_format = 'sdv-id-??????'
                     if is_numeric:
-                        bothify_format = '#########'
+                        column_dtype = str(column_dtype).lower()
+                        if 'int8' in column_dtype:
+                            bothify_format = '##'
+                        elif 'int16' in column_dtype:
+                            bothify_format = '####'
+                        else:
+                            bothify_format = '#########'
```

Comment:
I had to make this change to fix the dtype benchmarking because NumPy 2.0 raises an overflow error when you want to convert object numbers that exceed the value range of the dtype. Before, the behavior was weird; it was modifying the values of the number to be within the range allowed by the dtype.
Let me know if it makes sense.

Reply:
This makes sense. The only issue I see is that we're cutting the range of possible values to less than what it can be (eg. [-99, 99] instead of [-128, 127]. Maybe we should handle this in the place where we actually cast the string to an int?